### PR TITLE
Add checkbox component

### PR DESCRIPTION
**Pull request by igorschoester**
## Summary

This PR can be summarized in the following changelog entry:

* [@yoast/components] Updates the `Checkbox` component to the newer style. Now wrapped in a `FieldGroup`.
* [@yoast/components] Adds a maximum width to the `Alert` component.

## Relevant technical choices:

* Removing the old style checkbox (I don't see it being used), the props/usage remains the same 
* Using the field group as wrapper. This is suboptimal because: 
   * the BEM `__checkbox` should be a child of the field group
   * the field group is really more a fieldset, but then this component can only be singular
* Using `useCallback` from `React` here, but in the plugin that would've been from `@wordpress/element`. Should look into this sometime. 

## Test instructions

This PR can be tested by following these steps:

* Tested with News PR: https://github.com/Yoast/wpseo-news/pull/678

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Partially fixes https://yoast.atlassian.net/browse/P1-301

### CODE REVIEW

File: packages/components/src/checkbox/Checkbox.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from "prop-types";
+import React, { useCallback } from "react";
+import { FieldGroup } from "../field-group";
+import "./checkbox.css";
+
+/**
+ * Represents the input type checkbox with a label. Optionally a FieldGroup.
```
Comment (JoseeWouters) [inline packages/components/src/checkbox/Checkbox.js:7]: Please remove `optionally a fieldgroup` or make it optional.

**Review by JoseeWouters — CHANGES_REQUESTED** [review]
Please create a technical debt issue for the use of `useCallback` and one minor change for a comment.

**Review by JoseeWouters — APPROVED** [review]
CR 👍

**Comment by marijnyoast** [comment]
Acceptance 👍🏻

**Pull request merged**